src/components/MainTable/index.js

```diff
@@ -0,0 +1,188 @@
+                    return {...item, isPinned: !item.isPinned}
+                } else {
+                    return item
+                }
+            })
+            const pinnedArr = filteredArr.filter(item => item.isPinned === true)
+
+            dispatch(showDisplayedCountries(filteredArr.filter(item => item.isPinned === false)))
+            sessionStorage.setItem('display', JSON.stringify(filteredArr.filter(item => item.isPinned === false)))
+
+            dispatch(pin(pinnedArr[0]))
+            sessionStorage.setItem('pinned', JSON.stringify(JSON.parse(sessionStorage.getItem('pinned')).concat(pinnedArr)))
+
+        } else if (country.isPinned === true) {
+            const filteredArr = pinnedCountries.map((item) => {
+                if (item.id === id) {
+                    return {...item, isPinned: !item.isPinned,}
+                } else {
+                    return item
+                }
+            })
+            const unpinnedArr = filteredArr.filter(item => item.isPinned === false)
+            displayedCountries.unshift(unpinnedArr[0])
+
+            dispatch(showDisplayedCountries(displayedCountries))
+            sessionStorage.setItem('display',JSON.stringify(displayedCountries))
+
+            dispatch(unpin(filteredArr.filter(item => item.isPinned === true)))
+            sessionStorage.setItem('pinned', JSON.stringify(filteredArr.filter(item => item.isPinned === true)))
+        }
+    }
+
+    const deleteHandler = (e, id) => {
+        e.preventDefault()
+        const displayedFilteredArr = displayedCountries.filter(item => item.id !== id)
+        const pinnedFilteredArr = pinnedCountries.filter(item => item.id !== id)
+
+        dispatch(deleteFromDisplay(displayedFilteredArr))
+        sessionStorage.setItem('display', JSON.stringify(displayedFilteredArr))
+
+        dispatch(deleteFromPinned(pinnedFilteredArr))
+        sessionStorage.setItem('pinned', JSON.stringify(pinnedFilteredArr))
+    }
+
+    const dragStartHandler = (e, country) => {
+        setCurrentCard(country)
+    }
+
+    const dragOverHandler = e => {
+        e.preventDefault()
+    }
+
+    const dropHandler = (e, country) => {
+        e.preventDefault()
+        dispatch(showDisplayedCountries(displayedCountries.map( item => {
+            if (item.id === country.id) {
+                return {...item, order: currentCard.order}
+            }
+            if (item.id === currentCard.id) {
+                return {...item, order: country.order}
+            }
+            return item
+        })))
+    }
+
+    const sortCards = (a, b) => {
+        if (a.order > b.order) {
+            return 1
+        } else {
+            return -1
+        }
+    }
+
+
+    return (
+        <div className='main-wrapper'>
+            <div className='input-wrapper'>
+                <CustomInput 
+                    placeholder='Enter country name..'
+                    value={input}
+                    onChange={e => onChangeHandler(e)}
```

Comment:
It is better to avoid using arrow function in render method  you could write onChange={onChangeHandler} with the same result